### ISSUE TITLE
Uncaught TypeError: i.register is not a function

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#8615]
Vaadin 14.2.1 and 14.1.28, on a Maven-built Vaadin project. I'm not able to attach the project since I fail to reproduce the issue on my local machine. 

![stacktrace](https://user-images.githubusercontent.com/254091/85582536-2f8da780-b646-11ea-8486-36e7f5baaa45.png)

It would be good to have a set of steps to verify what could go wrong in this case. For example, it would be good to have a flow version with unobfuscated GWT code, or webpack with unminified code, so that it would be possible to investigate what's going on.

**Comment by mshabarov** [comment]
@mvysny could you please describe the circumstances and context of this error? I just want to estimate the severity and impact of this bug.

**Comment by mvysny** [comment]
The issue unfortunately only manifests under a very specific circumstances at the customer's machine. I was unable to reproduce the very same issue on my machine. Therefore the cause is a mystery.

I hoped that another person affected by this issue would be able to provide a reproducible example, but so far unfortunately that hasn't happened.

Maybe we could provide a special version of flow with unobfuscated gwt code; we could then perhaps include such a library into the customer's code and learn more from an unobfuscated stack trace?

**Comment by andrelgirao22** [comment]
I've been looking about this error, this may caused by the component license.

**Comment by mvysny** [comment]
I've managed to reproduce the issue. Steps to reproduce:


1. git clone https://github.com/mvysny/vaadin-coroutines-demo/
2. `./gradlew clean build -Pvaadin.productionMode`
3. run build/libs/vaadin-coroutines-demo.war with jetty-runner.

Alternatively:

1. navigate to https://vaadin-coroutines-demo.herokuapp.com/

**Comment by Artur-** [comment]
The error on https://vaadin-coroutines-demo.herokuapp.com/ comes from 
```
  var i = document.createElement("dom-module");
    i.id = "vcf-progress-spinner-lumo",
    i.setAttribute("theme-for", "vcf-progress-spinner"),
    i.innerHTML = "\n    <template>\n      <style>\n        :host {}\n      </style>\n    </template>\n  ",
    i.register(i.id)
}
```

Would be easier to see the origin if the bundle was not obfuscated

**Comment by Artur-** [comment]
Presumably from https://github.com/vaadin-component-factory/vcf-progress-spinner/blob/master/theme/lumo/vcf-progress-spinner-styles.js which seems to assume `dom-module` is a registered custom element, but in this case it is not, hence the `register` method is not found

**Comment by mvysny** [comment]
This is excellent pointer, thank you! Looks to me like a missing import in the vcf-progress-spinner-styles.js.

@Artur- would you happen to know a way to disable minification when calling webpack in production mode? That could help to figure out issues like these a lot.

**Comment by mvysny** [comment]
Looks like setting the following in `webpack.config.js` does the trick:

```js
module.exports = merge(flowDefaults, {
    optimization: {
        minimize: false
    }
});
```

Let me try again :+1:

**Comment by mvysny** [comment]
You're right, the deobfuscated code clearly points towards the vcf-progress-spinner (snippet from `http://localhost:8080/VAADIN/build/vaadin-bundle-2be600b9349182885bea.cache.js`):

```js
/***/ }),
/* 187 */,
/* 188 */
/***/ (function(module, exports) {

var theme = document.createElement('dom-module');
theme.id = 'vcf-progress-spinner-lumo';
theme.setAttribute('theme-for', 'vcf-progress-spinner');
theme.innerHTML = `
    <template>
      <style>
        :host {}
      </style>
    </template>
  `;
theme.register(theme.id);
```

**Comment by mvysny** [comment]
Filed https://github.com/vaadin-component-factory/vcf-progress-spinner/issues/3

Closing since the issue is not related to flow.
